Remora.Commands/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -25,236 +25,223 @@
+    /// <param name="treeName">The name of the tree to configure.</param>
+    /// <returns>A builder type that can be used to configure the tree.</returns>
+    public static TreeRegistrationBuilder AddCommandTree
+    (
+        this IServiceCollection serviceCollection,
+        string? treeName = null
+    )
     {
-        /// <summary>
-        /// Adds a command module to the available services.
-        /// </summary>
-        /// <typeparam name="TCommandModule">The command module to register.</typeparam>
-        /// <param name="serviceCollection">The service collection.</param>
-        /// <returns>The service collection, with the configured modules.</returns>
-        public static IServiceCollection AddCommandGroup<TCommandModule>
-        (
-            this IServiceCollection serviceCollection
-        )
-            where TCommandModule : CommandGroup
-        {
-            return serviceCollection.AddCommandGroup(typeof(TCommandModule));
-        }
+        serviceCollection.Configure<CommandTreeAccessorOptions>(o => o.AddTreeName(treeName));
+        return new TreeRegistrationBuilder(treeName, serviceCollection);
+    }
 
-        /// <summary>
-        /// Adds a command module to the available services.
-        /// </summary>
-        /// <param name="serviceCollection">The service collection.</param>
-        /// <param name="commandModule">The command module to register.</param>
-        /// <returns>The service collection with the configured modules.</returns>
-        public static IServiceCollection AddCommandGroup
+    /// <summary>
+    /// Adds a command module to the available services.
+    /// </summary>
+    /// <typeparam name="TCommandModule">The command module to register.</typeparam>
+    /// <param name="serviceCollection">The service collection.</param>
+    /// <returns>The service collection, with the configured modules.</returns>
+    [Obsolete("Call AddCommandTree with no arguments instead, and add the group there.")]
+    public static IServiceCollection AddCommandGroup<TCommandModule>
+    (
+        this IServiceCollection serviceCollection
+    ) where TCommandModule : CommandGroup
+        => serviceCollection.AddCommandGroup(typeof(TCommandModule));
+
+    /// <summary>
+    /// Adds a command module to the available services.
+    /// </summary>
+    /// <param name="serviceCollection">The service collection.</param>
+    /// <param name="commandModule">The command module to register.</param>
+    /// <returns>The service collection with the configured modules.</returns>
+    [Obsolete("Call AddCommandTree with no arguments instead, and add the group there.")]
+    public static IServiceCollection AddCommandGroup
+    (
+        this IServiceCollection serviceCollection,
+        Type commandModule
+    )
+    {
+        return serviceCollection
+            .AddCommandTree()
+            .WithCommandGroup(commandModule)
+            .Done();
```

Comment:
I've yet to see what this method does, but wouldn't `.Complete()` or `.Finalize()` be more apt of a name?